lib/miq_expression.rb

```diff
@@ -1360,26 +1360,9 @@ def self.get_column_details(column_names, class_path, assoc_path, opts)
   end
 
   def self.get_col_type(field)
-    model, parts, col = parse_field(field)
-
-    return :string if model.downcase == "managed" || parts.last == "managed"
-    return nil unless field.include?("-")
-
-    model = determine_model(model, parts)
-    return nil if model.nil?
-    return Field.parse(field).column_type if col.include?(CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX)
-
-    Field.new(model, [], col).column_type
-  end
-
-  def self.parse_field(field)
-    col = field.split("-").last
-    col = col.split("__").first unless col.nil? # throw away pivot table suffix if it exists before looking up type
-
-    parts = field.split("-").first.split(".")
-    model = parts.shift
-
-    return model, parts, col
+    parse_field_or_tag(field).column_type
+  rescue MiqExpression::Field::ParseError
```

Comment:
I know we have talked about this before, but does it make sense to have (for badly formed fields) `Field.parse # => nil` and `Field.parse! # => ParseError`?